tests/main/snap-handle-link/task.yaml

```diff
@@ -1,33 +1,30 @@
 summary: Ensure the "snap handle-link" command works
 
-# We don't run on ubuntu-core, since we are testing the behaviour of
-# xdg-open on the host system.
-systems: [-ubuntu-core-*]
-
-# xdg-open doesn't seem to check desktop files if it doesn't think it
-# has a display.
-environment:
-    DISPLAY: :placeholder
+systems:
+    - -amazon-linux-2-*  # does not support systemd --user
+    - -centos-7-*  # does not support systemd --user
+    - -ubuntu-14.04-*  # not supported as desktop OS, systemd too old for tests.session
+    - -ubuntu-core-*  # test exercises regular xdg-open, not the custom one in ubuntu-core
 
 prepare: |
     touch /usr/bin/zenity
+    tests.session -u test prepare
+
+    # Make sure there is a default handler to open a snap package
+    tests.session -u test exec mkdir -p /home/test/.config
+    tests.session -u test exec xdg-mime default snap-handle-link.desktop x-scheme-handler/snap
```

Comment:
this should be unset/removed in restore. I believe backing up `~test/.config/mimeapps.list` and restoring it in restore should be sufficient.

Reply:
Done